pkg/graph/resolvers/itgovactions/newstep/progress_to_new_step.go

```diff
@@ -1,191 +1,49 @@
-			if errRequesterFeedback != nil {
-				return errRequesterFeedback
-			}
-
-			return nil
-		})
-	}
-
-	// save feedback/recommendations for GRB
-	if input.GrbRecommendations != nil {
-		errGroup.Go(func() error {
-			feedbackForGRB := &models.GovernanceRequestFeedback{
-				IntakeID:     intake.ID,
-				Feedback:     *input.GrbRecommendations,
-				SourceAction: models.GRFSAProgressToNewStep,
-				TargetForm:   models.GRFTFNoTargetProvided,
-				Type:         models.GRFTGRB,
-			}
-			feedbackForGRB.CreatedBy = adminEUAID
-
-			_, errGRBFeedback := store.CreateGovernanceRequestFeedback(ctx, feedbackForGRB)
-			if errGRBFeedback != nil {
-				return errGRBFeedback
-			}
-
-			return nil
-		})
-	}
-
-	// save admin note
-	if input.AdminNote != nil {
-		errGroup.Go(func() error {
-			adminUserInfo, errAdminUserInfo := fetchUserInfo(ctx, adminEUAID)
-			if errAdminUserInfo != nil {
-				return errAdminUserInfo
-			}
-
-			adminNote := &models.SystemIntakeNote{
-				SystemIntakeID: input.SystemIntakeID,
-				AuthorEUAID:    adminEUAID,
-				AuthorName:     null.StringFrom(adminUserInfo.CommonName),
-				Content:        null.StringFromPtr(input.AdminNote),
-			}
-
-			_, errCreateNote := store.CreateSystemIntakeNote(ctx, adminNote)
-			if errCreateNote != nil {
-				return errCreateNote
-			}
-
-			return nil
-		})
-	}
-
-	if err := errGroup.Wait(); err != nil {
-		return nil, err
-	}
-
-	return updatedIntake, nil
-}
-
-// the frontend shouldn't allow users to take any invalid actions, but validate server-side to make sure
-func isIntakeValid(intake *models.SystemIntake, newStep model.SystemIntakeStepToProgressTo) error {
+// IsIntakeValid checks if in an intake and a new step from a Progress to New Step action are valid
+// the frontend shouldn't allow users to take any invalid actions, but we validate server-side to make sure
+func IsIntakeValid(intake *models.SystemIntake, newStep model.SystemIntakeStepToProgressTo) error {
 	if intake.State == models.SystemIntakeStateCLOSED {
-		return &apperrors.InvalidActionError{
-			ActionType: models.ActionTypePROGRESSTONEWSTEP,
-			Message:    "Can't take Progress to New Step action on closed intakes",
+		return &apperrors.BadRequestError{
+			Err: &apperrors.InvalidActionError{
+				ActionType: models.ActionTypePROGRESSTONEWSTEP,
+				Message:    "Can't take Progress to New Step action on closed intakes",
+			},
 		}
 	}
 
 	if intake.Step == models.SystemIntakeStepINITIALFORM && intake.RequestFormState == models.SIRFSNotStarted {
-		return &apperrors.InvalidActionError{
-			ActionType: models.ActionTypePROGRESSTONEWSTEP,
-			Message:    "Can't take Progress to New Step action on intakes that haven't started the Request Form",
+		return &apperrors.BadRequestError{
+			Err: &apperrors.InvalidActionError{
+				ActionType: models.ActionTypePROGRESSTONEWSTEP,
+				Message:    "Can't take Progress to New Step action on intakes that haven't started the Request Form",
```

Comment:
I don't think the compiler will catch these because you aren't `using fmt.ErrorOf()`. I think this error, and the error below would ideally start with a lower case character to follow the go paradigm of error casing.

Reply:
Good point, one moment.

Reply:
Decided not to change this for right now, mainly so I can get this PR done with today. If we do change it later, we'll also probably want to change the `.Error()` implementations in `pkg/apperrors/errors.go`, which all start with upper case letters.